fennel/testing/query_engine.py

```diff
@@ -346,6 +353,30 @@ def _check_schema_exceptions(
                 f"invalid schema for data: {exceptions}"
             )
 
+    def _compute_expr_extractor(
+        self,
+        extractor: Extractor,
+        intermediate_data: Dict[str, pd.Series],
+    ) -> pd.Series:
+        if len(extractor.outputs) != 1:
+            raise ValueError(
+                f"Expr extractor {extractor.name} must have exactly one output feature, found {len(extractor.outputs)}"
+            )
```

Comment:
`extractor for feature {} outputs {len(extractor.outputs)} features, expected one`